### ISSUE TITLE
Footmisc hang option in combination with bigfoot

### ISSUE THREAD

**Issue by ChristianSonderUniSG** [FrankMittelbach/fmitex-footmisc#14]
There exists an issue with the "hang" option of footmisc when the bigfoot package is loaded.
This combination produces empty lines after each footnote.

Compare: 
https://tex.stackexchange.com/questions/715860/footmisc-hanging-option-produces-empty-lines-when-bigfoot-is-loaded
and: 
https://tex.stackexchange.com/questions/242161/odd-vertical-space-combining-footmisc-with-hang-option-and-bigfoot?noredirect=1&lq=1

Ca. 9 years ago, Ulrike Fischer analysed, that there is a "\par  too much in this combination",
but applying Ulrikes solution makes the footnotemarks to appear left of the type area.

@FrankMittelbach: Do you see a possibility to fix this issue?

**Comment by FrankMittelbach** [comment]
Looks like bigfoot adds a \par at the end of the footnote text, ie as uf you had said `\footnote{Some text.\par}`.

Core LaTeX (since more than 17 years ago) and footmisc add a strut at the end of that text to keep the footnote separation consistent. But now this strut is on a line by itself, hence the sudden extra space (which is equal to a text line without visible content.

Try this, but I don't know if it break tabular or other places ... no time to check that right now.
```
\makeatletter
\def\@finalstrut#1{%
  \unskip
  \ifhmode \nobreak \vrule
  \else
    \ifdim \prevdepth=-\@m\p@
    \else
      \vskip-\prevdepth
    \fi
    \hrule
  \fi
  \@width\z@\@height\z@\@depth\dp#1}
\makeatother
```
@davidcarlisle @u-fischer  any thoughts about that? The core 2e definition is a somewhat primitive and simply assumes that we are in hmode really.

**Comment by davidcarlisle** [comment]
@FrankMittelbach  I know it was like that before, but shouldn't the unskip be inside the hmode test? Otherwise it's discarding any explicit vertical space at the end of the footnote, I suppose it simplifies the prevdepth test but if you end a footnote with `\par\vspace{1cm}` shouldn't you get 1cm space?

**Comment by u-fischer** [comment]
@davidcarlisle  I don't see that the space is lost.

**Comment by davidcarlisle** [comment]
perhaps but `*` guards space from being dropped from the top not from being removed by unskip

**Comment by FrankMittelbach** [comment]
Doesn't it add zero space at the end to survive a single unskip precisely for that reason?

**Comment by FrankMittelbach** [comment]
in fact this \@zeroskip is already in \vspace which explains why the \vspace doesn't go away  as @u-fischer observed.

**Comment by ChristianSonderUniSG** [comment]
Would'nt it be cleaner to check whether bigfoot is loaded? E. g. to replace lines 525-527 with:
``` 
    \ifFN@hangfoot
      \@ifpackageloaded{bigfoot}{%
        \egroup % no \par if bigfoot is loaded, because bigfoot adds \par to all footnotes.
      }{%
        \par\egroup
      }
    \fi
```
instead of 
```
    \ifFN@hangfoot
      \par\egroup
    \fi
```

This would solve the extra \par immediately. But still there is the problem, that the footnotemark land in the left margin instead of being flushed with the margin.

**Comment by FrankMittelbach** [comment]
@ChristianSonderUniSG well, it doesn't solve the problem, does it? That is it shifts the hanging whereas my patch does (which also means that bigfoot does something slightly different to `\footnote{..\par}` as I claimed, probably involving some `\aftergroup` hackery).

But in any case, I don't think footmisc should fix a problem introduced by another package through testing each an every time if this or that package is loaded. So I don't really think that would be "cleaner".

My suggested fix is really a fix for core LaTeX as it indeed is making its life there a little bit too easy, which is why I think that is the more correct approach. I just ran it against the full testsuite and and it shows only one difference (which was actually showing that the old definition was wrong, only it was overlooked).
